File: TestNG/Activities/Activity3.java
package activities;
import java.time.Duration;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity3 {
    WebDriver driver;
    WebDriverWait wait;

    @BeforeClass
    public void beforeClass() {
        driver = new FirefoxDriver();
        wait = new WebDriverWait(driver,Duration.ofSeconds(10));
        // Open browser
        driver.get("https://training-support.net/webelements/login-form");
    }

    @Test
    public void loginTest() {
        // Find the username and password fields
        WebElement username = driver.findElement(By.id("username"));
        WebElement password = driver.findElement(By.id("password"));

        // Enter credentials
        username.sendKeys("admin");
        password.sendKeys("password");

        // Click login
        driver.findElement(By.xpath("//button[text()='Submit']")).click();

        // Read login message

        //String loginMessage = driver.findElement(By.cssSelector("h2.text-center")).getText();
        String loginMessage = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("h2.mt-5"))).getText();
        Assert.assertEquals("Welcome Back, Admin!", loginMessage);
        String loginMessage2 = driver.findElement(By.cssSelector("h2.mt-5")).getText();
        Assert.assertEquals("Welcome Back, Admin!", loginMessage2);
        
    }

    @AfterClass
    public void afterClass() {
        // Close browser
        driver.close();
    }
}

File: Python/Activities/Activity3.py
from selenium import webdriver
from selenium.webdriver.common.by import By

# Start the Driver
driver = webdriver.Firefox()
driver.get("https://training-support.net/webelements/login-form")

print("Page title is:", driver.title)

# Find fields
username = driver.find_element(By.ID, "username")
password = driver.find_element(By.ID, "password")

# Enter credentials
username.send_keys("admin")
password.send_keys("password")

# Click login
login = driver.find_element(By.XPATH, "//button[text()='Submit']")
login.click()

# Print login message
message = driver.find_element(By.XPATH, "//h1[contains(@class, 'text-center')]")
print("Login message:", message.text)

driver.quit()
